### PR TITLE
Add option to use term colors.

### PR DESCRIPTION
**Pull request by beauby**
Ref: #18.

### CODE REVIEW

**Comment by bbatsov** [comment]
Can you show us how the degraded palette looks?

File: solarized.el [hunk 1]
```diff
@@ -116,6 +116,11 @@ Related discussion: https://github.com/bbatsov/solarized-emacs/issues/158"
   :type 'boolean
   :group 'solarized)
 
+(defcustom solarized-termcolors nil
```
Comment (bbatsov) [inline solarized.el]: Shouldn't we detect the capabilities of the terminal?

Comment (beauby) [reply solarized.el]: AFAIK, the main issue is that a 256 color capable term emulator will be recognized as such, and those 256 colors do not contain the exact solarized palette, although the terminal might be configured with the official 16 colors palette as base colors. I may be wrong though.

Comment (thomasf) [reply solarized.el]: Yes, detection should not be trusted as the single source of truth.

Comment (bbatsov) [reply solarized.el]: I see.

File: solarized.el [hunk 2]
```diff
@@ -126,11 +131,13 @@ COLOR1 and COLOR2 should be color names (e.g. \"white\") or RGB
...
 
 Alpha should be a float between 0 and 1."
-  (apply 'color-rgb-to-hex
-         (-zip-with '(lambda (it other)
-                       (+ (* alpha it) (* other (- 1 alpha))))
-                    (color-name-to-rgb color1)
-                    (color-name-to-rgb color2))))
+  (if solarized-termcolors
+      (if (< alpha 0.5) color2 color1)
+      (apply 'color-rgb-to-hex
```
Comment (bbatsov) [inline solarized.el]: This doesn't look right - the `else` should be outdented.

File: solarized.el [hunk 3]
```diff
@@ -126,11 +131,13 @@ COLOR1 and COLOR2 should be color names (e.g. \"white\") or RGB
 triplet strings (e.g. \"#ff12ec\").
 
 Alpha should be a float between 0 and 1."
-  (apply 'color-rgb-to-hex
-         (-zip-with '(lambda (it other)
-                       (+ (* alpha it) (* other (- 1 alpha))))
-                    (color-name-to-rgb color1)
-                    (color-name-to-rgb color2))))
+  (if solarized-termcolors
+      (if (< alpha 0.5) color2 color1)
```
Comment (bbatsov) [inline solarized.el]: I'm not quite certain what this is supposed to do. Some comments might be useful.

**Comment by 0e39bf7b** [comment]
@bbatsov what should we to to merge this PR?

File: solarized.el (hunk 1 above)
Comment (luisgerhorst) [reply solarized.el]: Is there any widespread terminal that correctly displays the solarized color codes? If not what about `(not (display-graphic-p)` as a heuristic? Either the user has set the solarized palette in their terminal or he did not. 
- If he has set the solarized palette it works fine if we use the heuristic. If we don't they look bad.
- If he has not set the solarized palette the colors look bad anyway, so it does not make a difference if we select the "fake" palette using the heuristic (assuming nobody wants to use the solarized theme in a 256 terminal where the colors are only displayed approximately).

**Comment by thomasf** [comment]
Closing this because not compatible with current code.

This does not solve all use of color blending in the theme. Everything not using the standard 16 solarized colors needs to have individually tested for every change. 

I think it's just too much work given that there are 24bit compatible terminals that work in macos, linux and windows people should be nudged towards learning how to set that up instead.

**Pull request closed without merging**